src/expression_builder.rs

```diff
@@ -17,45 +17,42 @@ pub enum Operand<TS: TypeSystem> {
     Expression(Box<ExpressionBuilder<TS>>),
 }
 
-#[derive(Default, Clone)]
+#[derive(Default, Clone, Debug)]
 pub struct ExpressionBuilder<TS: TypeSystem> {
     operator: Option<Operator<TS>>,
     operands: (Option<Operand<TS>>, Option<Operand<TS>>),
```

Comment:
Why are we still holding onto these fields when they are just passed directly as arguments to our functions? This setup, with 3 options, allows for invalid states to be represented - we should just turn this into an Expression enum or something, and have a method to convert any given variant to a list of instructions.

Reply:
Originaly kept it for the simplicity of the refactor, didnt get around to going back and changing it yet because I was focused on other aspects